cmake/helpers.cmake

```diff
@@ -244,8 +244,9 @@ function(add_umf_target_compile_options name)
             target_compile_definitions(${name} PRIVATE -D_FORTIFY_SOURCE=2)
         endif()
         if(UMF_DEVELOPER_MODE)
-            target_compile_options(${name} PRIVATE -fno-omit-frame-pointer
-                                                   -fstack-protector-strong)
+            target_compile_options(
+                ${name} PRIVATE -fno-omit-frame-pointer
+                                -fstack-protector-strong -Werror)
```

Comment:
please add similar flag for windows and clang (check for any warnings left in CI)